src/pages/Profile/index.jsx

```diff
@@ -49,45 +51,77 @@ function Profile() {
   const handleClose = () => setOpen(false);
 
   const handleSendFriendRequest = () => {
-    const currentUserUid = auth.currentUser.uid;
-    const targetUserUid = currentUserUid; // TODO: Change this to the user whose profile is being viewed
-    const friendRequestData = {
-      sender: currentUserUid,
-      recipient: targetUserUid,
-      timestamp: firebase.firestore.FieldValue.serverTimestamp(),
-    };
-    db.collection("friendRequests")
-      .add(friendRequestData)
-      .then(() => {
-        setFriendRequestSent(true);
-        playSuccessSound();
-        enqueueSnackbar("Friend request sent!", {
-          variant: "success",
+    const currentUser = auth.currentUser;
+    const currentUserUid = currentUser.uid;
+    const targetUserUid = uid;
+    if (friendRequestSent) {
+      db.collection("users")
+        .doc(targetUserUid)
+        .collection("friendRequests")
+        .doc(currentUserUid)
+        .delete()
+        .then(() => {
+          db.collection("users")
+            .doc(targetUserUid)
+            .collection("notifications")
+            .doc(currentUserUid)
+            .delete()
+            .then(() => {
+              setFriendRequestSent(false);
+            });
```

Comment:
add a snackbar telling the status of deleting friendrequest

Reply:
Yup I've added!